.github/workflows/premerge.yml

```
@@ -0,0 +1,57 @@
+name: Premerge
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+  schedule:
+    - cron: 0 0 1 * *
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2020-09-27
+          override: true
+
+      - name: Build check
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+
+      - name: Tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+  linting:
+    if: ${{ github.ref != 'main' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
```

```suggestion
          toolchain: nightly-2020-09-27
```

It's probably respecting the `rust-toolchain` file when you run fmt, when though you specify the nightly toolchain

---

Yeah, I had a bit of a moment here, sorry for the push spam 😅

---

It will respect (I tested) but it will redownload potentially another toolchain. On my machine it download that specific toolchain automatically